Source/EasyNetQ/Consumer/DefaultConsumerErrorStrategy.cs

```diff
@@ -200,12 +210,20 @@ private string CreateConnectionCheckMessage()
         }
 
         private bool disposed = false;
+        private bool disposing = false;
 
         public virtual void Dispose()
         {
             if (disposed) return;
+            disposing = true;
 
-            if(connection != null) connection.Dispose();
+            lock (syncLock)
```

Comment:
actually I don't like the idea of having locks in dispose, could we discuss a little bit more on this?